sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -176,10 +176,10 @@ private boolean canGatewayRequestFailoverOnTimeout(RxDocumentServiceRequest requ
               return true;
         }
 
-        //Data Plane Read
+        //Data Plane Read and Writes
         if(!isMetaDataRequest
             && !request.isAddressRefresh()
-            && request.isReadOnly()) {
+            && (request.isReadOnly()) || !BridgeInternal.hasSendingRequestStarted(clientException)) {
```

Comment:
`canGatewayRequestFailoverOnTimeout ` is only called for `HttpConstants.SubStatusCodes.GATEWAY_ENDPOINT_READ_TIMEOUT`

I think `HttpConstants.SubStatusCodes.GATEWAY_ENDPOINT_READ_TIMEOUT` will only be set when the request has been written to backend and SDK failed to get response back, so which is already an indication can not retry safely for write operations.

Reply:
Ok, so since if data plane writes can only be retried on connection issues and when it does have a connection issue it will have a `GATEWAY_ENDPOINT_UNAVAILABLE` sub status code and that is already handled, should we close this PR? Or should we still move forward with wiring the `hasSendingRequestStartedFlag`?

Reply:
I think we should close this PR. @kushagraThapar @FabianMeiswinkel would like to get your thoughts as well